frontend-svelte/src/routes/+page.svelte

```diff
@@ -1,16 +1,49 @@
 <script>
 	import { base } from '$app/paths';
 	import { Button } from '$lib/components/ui/button/';
+	import { goto } from '$app/navigation';
+	import bgImg3 from '$lib/assets/bg/3.jpeg';
 
-  console.log(`base is ${base}`)
-</script>
+  	console.log(`base is ${base}`)
+
+  	let { data } = $props();
 
-<div class="container">
-	<h1 class="font-accent italic text-5xl my-4">fabX</h1>
-	<h2 class="font-accent text-2xl my-2">Access System for Makerspaces</h2>
+	$effect(() => {
+			// Wait for me data to be loaded by the server
+			const me = data.me;
+			// Handle redirection based on user type
+			if (me) {
+				if (me.isAdmin) {
+					goto(`${base}/admin`);
+				} else {
+					goto(`${base}/instructor`);
+				}
+			} else {
+				// No user data means not logged in
+				goto(`${base}/login`);
+			}
+			// TODO: Improve error handling
+			const error = data.error
```

Comment:
Nitpick: I don't think this needs assignment to a new variable, just use `data.error` twice 🤷